pulser-core/pulser/json/abstract_repr/deserializer.py

```diff
@@ -0,0 +1,157 @@
+        return BlackmanWaveform(obj["duration"], obj["area"])
+    if obj["kind"] == "blackman_max":
+        return BlackmanWaveform.from_max_val(obj["max_val"], obj["area"])
+    if obj["kind"] == "interpolated":
+        return InterpolatedWaveform(
+            obj["duration"], obj["values"], obj["times"]
+        )
+    if obj["kind"] == "kaiser":
+        return KaiserWaveform(obj["duration"], obj["area"], obj["beta"])
+    if obj["kind"] == "kaiser_max":
+        return KaiserWaveform.from_max_val(
+            obj["max_val"], obj["area"], obj["beta"]
+        )
+    if obj["kind"] == "composite":
+        wfs = [_deserialize_abstract_waveform(wf) for wf in obj["waveforms"]]
+        return CompositeWaveform(*wfs)
+    if obj["kind"] == "custom":
+        return CustomWaveform(obj["samples"])
+
+    raise AbstractReprError("The object does not encode a known waveform.")
+
+
+def _deserialize_abstract_operation(seq: Sequence, op: dict) -> None:
+    if op["op"] == "target":
+        seq.target_index(
+            qubits=op["target"],
+            channel=op["channel"],
+        )
+    elif op["op"] == "align":
+        seq.align(*op["channels"])
+    elif op["op"] == "delay":
+        seq.delay(
+            duration=op["time"],
+            channel=op["channel"],
+        )
+    elif op["op"] == "phase_shift":
+        seq.phase_shift_index(
+            op["phi"],
+            *op["targets"],
+        )
+    elif op["op"] == "pulse":
+        pulse = Pulse(
+            amplitude=_deserialize_abstract_waveform(op["amplitude"]),
+            detuning=_deserialize_abstract_waveform(op["detuning"]),
+            phase=op["phase"],
+            post_phase_shift=op["post_phase_shift"],
+        )
+        seq.add(
+            pulse=pulse,
+            channel=op["channel"],
+            protocol=op["protocol"],
+        )
+
+
+def deserialize_abstract_sequence(obj_str: str) -> Sequence:
+    """Deserialize a sequence from an abstract JSON object.
+
+    Args:
+        obj_str (str): the JSON string representing the sequence encoded
+            in the abstract JSON format.
+
+    Returns:
+        Sequence: The Pulser sequence.
+    """
+    pass
+
+    obj = json.loads(obj_str)
+
+    # Validate the format of the data against the JSON schema.
+    jsonschema.validate(instance=obj, schema=schema)
+
+    # Device
+    device_name = obj["device"]
+    device = getattr(devices, device_name)
+
+    # Register
+    qubits = obj["register"]
+    reg = Register({q["name"]: (q["x"], q["y"]) for q in qubits})
+
+    seq = pulser.Sequence(reg, device)
+
+    # Channels
+    for name, channel_id in obj["channels"].items():
+        seq.declare_channel(name, channel_id)
+
+    # Variables
+    vars = {}
+    for name, desc in obj["variables"].items():
+        v = seq.declare_variable(
+            cast(str, name),
+            size=len(desc["value"]),
+            dtype=VARIABLE_TYPE_MAP[desc["type"]],
+        )
+        vars[name] = v
```

Comment:
What's this dict for?

Reply:
This dict will be used for keeping a ref of all variables in order to use them during the deserilization of param objects. Not used yet though